envoy/ssl/handshaker.h

```diff
@@ -26,6 +27,9 @@ struct TlsContext;
 
 class ServerContextConfig;
 
+using CurveNID = int;
+using CurveNIDSupportedVector = absl::InlinedVector<int, 3>;
```

Comment:
```suggestion
// Currently this type only ever holds 3 values: P-256, P-384, and P-521, so optimize by using `InlinedVector`.
using CurveNIDVector = absl::InlinedVector<CurveNID, 3>;
```
Doesn't need `Supported` in the name; the datatype could hold curve NIDs for any reason.